### DISCUSSION TITLE
setFocus for input fields that are on another tab

### DISCUSSION THREAD

**Discussion by vladutclp** [react-hook-form/react-hook-form#10893] [Q&A]
Hi everyone, I have a big form that is split in 2 tabs. I have some validation and call `setFocus` to se the focus on the fields that have an error. When I press the submit button and the inputs are on the unmounted tab(the one that us not selected), I get this error:

![image](https://github.com/react-hook-form/react-hook-form/assets/22237268/a0fd5d4b-ecf2-4f09-97b9-3c3408f3db0a)
![image](https://github.com/react-hook-form/react-hook-form/assets/22237268/741a05c1-41df-4227-b22b-7eda425becbd)

I think the problem arises because the inputs are being unmounted and RHF is trying to `setFocus` on them, but I don't know how can I solve this.

Can someone give me a hint on this? 

Thank you!

**Comment by vuggy17** [comment]
Maybe you should find the tab which contains the error field before `setForcus`. 
The error says that is cannot read `focus` property of a input ref (which may be due to the ref being unmounted and therefore undefined).

**Comment by vladutclp** [comment]
@vuggy17 Thanks for the response. That's the solution I was thinking about and implemented but I thought there is another way of doing this.

**Comment by Moshyfawn** [comment]
If the input was never mounted (and its `ref` was never registered), there's no way for RHF to know about your input
